pkg/analysis/criteria/state_hash.go

```diff
@@ -21,32 +27,52 @@ type StateHashCriterion struct {
 func NewStateHashCriterion(es *events.Storage, opts *StateHashCriterionOptions) *StateHashCriterion {
 	if opts == nil { // default
 		opts = &StateHashCriterionOptions{
-			MaxForkDepth: 5,
+			MaxForkDepth:     defaultMaxForkDepth,
+			HeightBucketSize: defaultHeightBucketSize,
 		}
 	}
 	return &StateHashCriterion{opts: opts, es: es}
 }
 
 func (c *StateHashCriterion) Analyze(alerts chan<- entities.Alert, timestamp int64, statements entities.NodeStatements) error {
-	splitHeight := statements.SplitByNodeHeight()
-	for height, nodeStatements := range splitHeight {
-		if err := c.analyzeNodesOnSameHeight(alerts, height, timestamp, nodeStatements); err != nil {
-			return err
+	splitByBucketHeight := statements.SplitByNodeHeightBuckets(c.opts.HeightBucketSize)
+	for bucketHeight, nodeStatements := range splitByBucketHeight {
+		statementsAtBucketHeight := make(entities.NodeStatements, 0, len(nodeStatements))
+		for _, statement := range nodeStatements {
+			var statementAtBucketHeight entities.NodeStatement
+			if statement.Height == bucketHeight {
+				statementAtBucketHeight = statement
+			} else {
+				var err error
+				statementAtBucketHeight, err = c.es.GetStatementAtHeight(statement.Node, bucketHeight)
+				if err != nil {
+					if errors.Is(err, events.NoFullStatementError) {
+						// TODO: should we ignore statement in such case?
```

Comment:
The PR with zap logger has been merged, you can now add a log here

Reply:
Fixed